src/magicbox/ResultPreviewsManager.ts

```diff
@@ -0,0 +1,214 @@
+    return this.suggestionsPreviewContainer.findClass(this.options.previewClass);
+  }
+
+  private get suggestionsListbox() {
+    return $$($$(this.element).findClass('coveo-magicbox-suggestions')[0]);
+  }
+
+  private get numberOfResultsPerRow() {
+    const previewSelectables = this.suggestionsPreviewContainer.findClass(this.options.previewClass);
+    if (previewSelectables.length === 0) {
+      return 0;
+    }
+    const firstVerticalOffset = previewSelectables[0].offsetTop;
+    const firstIndexOnNextRow = findIndex(previewSelectables, previewSelectable => previewSelectable.offsetTop !== firstVerticalOffset);
+    return firstIndexOnNextRow !== -1 ? firstIndexOnNextRow : previewSelectables.length;
+  }
+
+  constructor(private element: HTMLElement, options: Partial<IResultPreviewsManagerOptions> = {}) {
+    this.options = defaults(options, <IResultPreviewsManagerOptions>{
+      previewHeaderText: l('QuerySuggestPreview'),
+      previewClass: 'coveo-preview-selectable',
+      selectedClass: 'magic-box-selected'
+    });
+    this.root = Component.resolveRoot(element);
+  }
+
+  public async displaySearchResultPreviewsForSuggestion(suggestion: HTMLElement) {
+    if (this.lastSelectedSuggestion !== suggestion) {
+      await this.loadSearchResultPreviews(suggestion);
+    }
+  }
+
+  public getElementInDirection(direction: Direction) {
+    const previewElements = this.previewElements;
+    const focusedIndex = previewElements.indexOf(this.focusedPreviewElement);
+
+    if (focusedIndex === -1) {
+      return null;
+    }
+
+    if (focusedIndex === 0 && direction === Direction.Left) {
+      return null;
+    }
+
+    return previewElements[(focusedIndex + this.getIncrementInDirection(direction)) % previewElements.length];
+  }
+
+  private getIncrementInDirection(direction: Direction) {
+    switch (direction) {
+      case Direction.Left:
+        return -1;
+      case Direction.Right:
+        return 1;
+      case Direction.Up:
+        return -this.numberOfResultsPerRow;
+      case Direction.Down:
+        return this.numberOfResultsPerRow;
+    }
+  }
+
+  private setHasPreviews(shouldHavePreviews: boolean) {
+    if (this.hasPreviews === !!shouldHavePreviews) {
+      return;
+    }
+    if (shouldHavePreviews) {
+      this.initPreviewForSuggestions();
+    } else {
+      this.revertPreviewForSuggestions();
+    }
+  }
+
+  private initPreviewForSuggestions() {
+    this.suggestionsPreviewContainer = $$(
+      'div',
+      {
+        className: 'coveo-suggestion-container'
+      },
+      this.suggestionsListbox.el,
+      this.buildPreviewContainer()
+    );
+    this.element.appendChild(this.suggestionsPreviewContainer.el);
+  }
+
+  private revertPreviewForSuggestions() {
+    console.log(this.suggestionsPreviewContainer);
+    console.log(this.hasPreviews);
+    this.element.appendChild(this.suggestionsListbox.el);
+    this.suggestionsPreviewContainer.remove();
+    this.suggestionsPreviewContainer = null;
+  }
```

Comment:
The first function wraps an already appended `suggestionsListbox` container in `coveo-suggestion-container` with the previews.
The second function unwraps it back to its original state.

This is only here for the sake of keeping the CSS intact along with any other component that might be checking the DOM structure before result previews are appended. It could be replaced with a `display: none;` and `display: block;` in the CSS.